fileServer.js

```diff
@@ -0,0 +1,48 @@
+ 'use strict';
+
+ var http = require('http');
+ var fs = require('fs');
+ var path = require('path');
+ var url = require('url');
+ var rootDir = __dirname + '/public';
+
+
+ const mimeType = {
+ 	'html': 'text/html',
+ 	'jpeg': 'image/jpeg',
+ 	'jpg': 'image/jpeg',
+ 	'png': 'image/png',
+ 	'js': 'text/javascript',
+ 	'css': 'text/css'
+ };
+
+ function send404Request(res) {
+ 	fs.readFile('./404.html', 'UTF-8', function(err, html) {
+ 		res.setHeader('Content-type', mimeType || 'text/html');
+ 		res.end(html);
+ 	});
+
+ }
+
+ function onRequest(req, res) {
+
+ 	var filename;
+ 	var pathname = url.parse(req.url).pathname;
+ 	filename = path.join(rootDir, pathname);
+ 	console.log('Filename: ' + filename);
+
+ 	if (req.method === 'GET' && req.url === '/') {
+ 		res.setHeader('Content-type', mimeType || 'text/html');
```

Comment:
You reference the `mimeType` hash but you do not access it with a key - so this code is not choosing a mime type based on the file extension.